test/goose/brokers/redis/batch_test.clj

```diff
@@ -40,12 +52,103 @@
           batch-state-key (d/prefix-batch batch-id)
           enqueued-job-set (d/construct-batch-job-set batch-id d/enqueued-job-set)
           job-ids (->>
-                    (map :id (:jobs batch))
-                    (set))]
+                   (map :id (:jobs batch))
+                   (set))]
       (is (= (redis-cmds/exists tu/redis-conn batch-state-key) 0))
       (is (= (redis-cmds/exists tu/redis-conn enqueued-job-set) 0))
 
       (redis-batch/enqueue tu/redis-conn batch)
 
       (is (= (redis-cmds/parse-map tu/redis-conn batch-state-key) batch-state))
-      (is (= (redis-cmds/set-members tu/redis-conn enqueued-job-set) job-ids)))))
\ No newline at end of file
+      (is (= (redis-cmds/set-members tu/redis-conn enqueued-job-set) job-ids)))))
+
+;;;; ======= TEST: Batch state update middleware ===========
```

Comment:
suggestion (non-blocking): I think it is worth checking set sizes of all the 4 sets in each of the tests, irrespective of the transition. Essentially, this would make the test more robust, checking for any unintended side-effects in the middleware.

For instance, in `enqueued-to-successful` we can add these test cases as well,
```clojure
(is (= 0 retrying-job-set-size))
(is (= 0 dead-job-set-size))
```